core/sched.c

```diff
@@ -218,6 +222,11 @@ thread_t *__attribute__((used)) sched_run(void)
             MPU_ATTR(1, AP_RO_RO, 0, 1, 0, 1, MPU_SIZE_32B) /* Attributes and Size */
             );
 #endif
+
+#ifdef MODULE_PMP_STACK_GUARD
+        write_pmpaddr(3, (uintptr_t)next_thread->stack_start);
```

Comment:
This magic number '3' is meh but I don't see how it can be done better in this PR. I will rework this soonish, which will also include the magic number '2' in the MPU case above.